### PR TITLE
Add board file for Pi-Plates MICROPi

### PR DESCRIPTION
**Pull request by pi-plates**
MICROpi board file for pico-sdk

### CODE REVIEW

**Review by peterharperuk — CHANGES_REQUESTED** [review]
You need to push this to the development branch, not master.

File: src/boards/include/boards/pi-plates_micropi.h [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ */
+
+// -----------------------------------------------------
+// NOTE: THIS HEADER IS ALSO INCLUDED BY ASSEMBLER SO
+//       SHOULD ONLY CONSIST OF PREPROCESSOR DIRECTIVES
+// -----------------------------------------------------
+
+// Board definition for the Pi-Plates MICROPi processor board
+//
+// This header may be included by other board headers as "boards/micropi.h"
```
Comment (lurch) [inline src/boards/include/boards/pi-plates_micropi.h:14]: The name here doesn't match the actual filename?

**Review by kilograham — APPROVED** [review]

**Comment by lurch** [comment]
With the following small tweaks, this passes the checks I added in #1724 
```patch
 
 // Board definition for the Pi-Plates MICROPi processor board
 
-#ifndef _BOARDS_PIPLATES_MICROPI_H_RP2040
-#define _BOARDS_PIPLATES_MICROPI_H_RP2040
+#ifndef _BOARDS_PI_PLATES_MICROPI_H
+#define _BOARDS_PI_PLATES_MICROPI_H
 
 // For board detection
-#define PIPLATES_MICROPI_RP2040
+#define PI_PLATES_MICROPI
 
 // --- UART ---
 #ifndef PICO_DEFAULT_UART
```

**Comment by kilograham** [comment]
this was added as part of #1724

**Comment by lurch** [comment]
> this was added as part of #1724

Umm, no it wasn't.

**Comment by kilograham** [comment]
is it ready to merge?

**Comment by lurch** [comment]
IMHO it'll be ready to merge if the changes I suggested earlier are made, i.e. 
```patch
 
 // Board definition for the Pi-Plates MICROPi processor board
 
-#ifndef _BOARDS_PIPLATES_MICROPI_H_RP2040
-#define _BOARDS_PIPLATES_MICROPI_H_RP2040
+#ifndef _BOARDS_PI_PLATES_MICROPI_H
+#define _BOARDS_PI_PLATES_MICROPI_H
 
 // For board detection
-#define PIPLATES_MICROPI_RP2040
+#define PI_PLATES_MICROPI
 
 // --- UART ---
 #ifndef PICO_DEFAULT_UART
```

Are you able to do that @pi-plates or should I just close this PR and create a new one? (Apologies for not making this clear in my previous comment.)

**Comment by pi-plates** [comment]
Im on VACA and will do this when I get back. I might need a little hand
holding though...

**Comment by lurch** [comment]
I've now made the above-mentioned changes directly.

**Review by lurch — APPROVED** [review]
`tools/check_board_header.py` is happy, so approving

**Comment by pi-plates** [comment]
Thanks a lot.
Jerry Wasinger
Founder, Pi-Plates

**Pull request merged**